signature/src/signer.rs

```diff
@@ -41,7 +43,7 @@ impl Ed25519DidSigner {
     }
 }
 
-impl DIDSigner<Ed25519Signature> for Ed25519DidSigner {
+impl DIDSigner<Ed25519Signature, crate::keypair::SSIKeyPair> for Ed25519DidSigner {
```

Comment:
Do you think its a problem that we have both a signature generic and a keypair generic? Do you think that'd cause problems?